src/bandersnatch/package.py

```diff
@@ -103,7 +104,9 @@ def save_json_metadata(self, package_info: Dict) -> bool:
         # In 4.0 we move to normalized name only so want to overwrite older symlinks
         if self.json_pypi_symlink.exists():
             self.json_pypi_symlink.unlink()
-        self.json_pypi_symlink.symlink_to(self.json_file)
+        self.json_pypi_symlink.symlink_to(
+            os.path.relpath(self.json_file, self.json_pypi_symlink.parent)
```

Comment:
I feel we could use the pathlib method and avoid importing os 
```suggestion
            self.json_file.relative_to(self.json_pypi_symlink.parent)
```

Reply:
Unfortunately pathlib.Path.relative_to() won't compute a relative path using a common prefix it only works if the argument to relative_to is a parent of the path, otherwise it raises a ValueError:
```
Traceback (most recent call last):
  File "relative_to.py", line 22, in <module>
    dest.relative_to(src)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/pathlib.py", line 899, in relative_to
    raise ValueError("{!r} does not start with {!r}"
ValueError: '/var/www/json/PKG' does not start with '/var/www/pypi/PKG/json'
```

Reply:
You're right - Disappointed that logic does not exist in pathlib - I might open a bug and add it, maybe with a slightly different API.

POC making it work: https://pastebin.com/zTVKFBam
https://pastebin.com/zTVKFBam